### PR TITLE
Fetch dependencies from conan if that is possible

### PR DESCRIPTION
**Pull request by KGrzeg**

### CODE REVIEW

**Comment by DoomHammer** [comment]
It looks like it could use a rebase

**Comment by KGrzeg** [comment]
That's better

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,9 @@ project(ohNet LANGUAGES CXX C)
 
 set(platform Posix)
 
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```
Comment (DoomHammer) [inline CMakeLists.txt]: The currently preferred way is to use Conan with find_package generator and use `find_packages` and `target_link_libraries` individually for each imported lib.

See for reference: https://github.com/PacktPublishing/Software-Architecture-with-Cpp/blob/master/Chapter08/customer/CMakeLists.txt#L21

File: default.nix [hunk 2]
```diff
@@ -14,5 +14,6 @@ mkShell {
     gcc
     gnumake
     ninja
+    # conan install manually the latest version!
```
Comment (DoomHammer) [inline default.nix]:
```
 shellHook = ''
    [ ! -d venv ] && python -m virtualenv venv
    source venv/bin/activate
    pip install conan==1.54.0
  '';
```

**Review by DoomHammer — CHANGES_REQUESTED** [review]
We should use the `CMakeDeps` Conan generator as the other one is deprecated.

**Review by DoomHammer — APPROVED** [review]

**Pull request merged**